frontend/packages/policy-editor/src/components/PolicyCardRules/PolicyRule/PolicyAccessPackages/PolicyAccordion/PolicyAccordion.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useId, useState } from 'react';
+import cn from 'classnames';
+import { StudioButton, StudioLabelAsParagraph } from '@studio/components';
+import * as StudioIcons from '@studio/icons';
+import { ChevronDownIcon, ChevronUpIcon } from '@studio/icons';
+import classes from './PolicyAccordion.module.css';
+
+interface PolicyAccordion {
+  icon?: string;
+  title: string;
+  subTitle: string;
+  extraHeaderContent?: React.ReactNode;
+  defaultOpen?: boolean;
+  onOpened?: () => void;
+  children: React.ReactNode;
+}
+
+export const PolicyAccordion = ({
+  icon,
+  title,
+  subTitle,
+  extraHeaderContent,
+  defaultOpen,
+  onOpened,
+  children,
+}: PolicyAccordion): React.ReactNode => {
+  const contentId = useId();
+  const [isExpanded, setIsExpanded] = useState<boolean>(defaultOpen || false);
+  const IconComponent = StudioIcons[icon];
+
+  useEffect(() => {
+    if (isExpanded && onOpened) {
+      onOpened();
+    }
+  }, [isExpanded, onOpened]);
+
```

Comment:
Can we make a hook of this? 😄

Reply:
moved code so this useEffect is no longer needed :D